src/main/kotlin/gg/essential/elementa/markdown/MarkdownComponent.kt

```diff
@@ -177,6 +179,21 @@ class MarkdownComponent(
         getTextScale(),
     )
 
+    fun onLinkClicked(block: MarkdownComponent.(LinkClickEvent) -> Unit) {
+        linkClickListeners.add(block)
+    }
+
+    internal fun fireLinkClickEvent(event: LinkClickEvent): Boolean {
+        for (listener in linkClickListeners) {
+            this.listener(event)
+
+            if (event.propagationStoppedImmediately) return false
+        }
+        return true
```

Comment:
We should probably also handle `event.propagationStopped`. I guess the expected behavior for that would be that all listeners get notified but then the url isn't opened  at the end in any case.